packages/cache-map/src/cachemap.js

```diff
@@ -132,6 +132,9 @@ const resetCell = (cell, oldKey, makeMap) => {
     if (size !== 1) {
       throw Error('internal: Unexpected non-singular cell data entry count');
     }
+    // Manually run the Iterator interface to ensure reading oldKey from an
+    // IteratorResult (rather than defaulting to undefined if the iteration is
+    // empty).
```

Comment:
Wouldn't the manual usage of the iterator result also get us `undefined` if the iteration is empty, as it's the likely value when we have a `done: true` result/